python/UniverseWrapper.cpp

```diff
@@ -471,6 +484,16 @@ namespace FreeOrionPython {
                                                     return_value_policy<return_by_value>()
                                                 ))
 
+            // Indexed by stat name (string), contains a map indexed by empire id,
+            // contains a map from turn number (int) to stat value (double).
```

Comment:
In this case the robustness of a map is preferable for in the C++ backend, since it's not guaranteed that all stats will be recorded on all turns... ie. a new stat could be scripted and added, or a stat script removed, or stat tracking turned on or off. If I use a list or vector that just stores the values without corresponding turns, then it would have to have an entry for every stat for every turn, including some way to store "missing value" sentinels, and I'd rather not need to deal with that complexity. 

For Python, using 0 or -1 for "no info" isn't very reliable, since many stats could return -1 or 0 as valid results. NaN is perhaps a better alternative, if Python supports it robustly.